Src/BootCamp.Chapter/Homework4.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BootCamp.Chapter
+{
+    class Homework4
+    {
+        public static void Demo()
+        {
+            static void GatherInfo(string userNumber)
+            {
+                Console.WriteLine("Hello! You are the " + userNumber + " user today.");
+                string firstName = PromptString("What is your first name?");
+                string surName = PromptString("What is your surname?");
+                int age = PromptInt("What is your age?");
+                float weight = PromptFloat("How much do you weigh(in kg)?");
+                float height = PromptFloat("How tall are you(in cm)?");
+
+                float BMI = CalculateBmi(weight, height);
+
+                PrintResult(firstName, surName, age, weight, height, BMI);
+            }
+            GatherInfo("first");
+            GatherInfo("second");
+        }
+        public static string PromptString(string message)
+        {
+            Console.WriteLine(message);
+            string Name = Console.ReadLine();
+
+            if (string.IsNullOrEmpty(Name))
```

Comment:
This should happen first and only then do the calculation. It's called "Fail Early" principle. makes programmer care about less things later down the line.